cannon/mipsevm/exec/calling_convention.go

```diff
@@ -0,0 +1,56 @@
+package exec
+
+// FYI: https://en.wikibooks.org/wiki/MIPS_Assembly/Register_File
+const (
+	Reg0  = 0
+	RegAt = 1
+	// syscall number; 1st return value
+	RegV0 = 2
+	// 2nd return value
+	RegV1 = 3
+	// syscall arguments; returned unmodified
+	RegA0 = 4
+	RegA1 = 5
+	RegA2 = 6
+	// 4th syscall argument; set to 0/1 for success/error
+	RegA3 = 7
+	// caller saved
+	RegT0 = 8
+	RegT1 = 9
+	RegT2 = 10
+	RegT3 = 11
+	RegT4 = 12
+	RegT5 = 13
+	RegT6 = 14
+	RegT7 = 15
+	// callee saved
+	RegS0 = 16
+	RegS1 = 17
+	RegS2 = 18
+	RegS3 = 19
+	RegS4 = 20
+	RegS5 = 21
+	RegS6 = 22
+	RegS7 = 23
+
+	RegT8 = 24
+	RegT9 = 25
+	RegK0 = 26
+	RegK1 = 27
+	RegGP = 28
+	RegSP = 29
+	RegFP = 30
+	RegRA = 31
+)
+
+// FYI: https://web.archive.org/web/20231223163047/https://www.linux-mips.org/wiki/Syscall
+const (
+	RegSyscallNum    = RegV0
+	RegSyscallRet1   = RegV0
+	RegSyscallRet2   = RegV1
+	RegSyscallResult = RegA3
```

Comment:
I think this register is for error results:
```suggestion
	RegSyscallError = RegA3
```

Reply:
Why is a3 called syscall errno/result? We only use it for `timeout` for futex

Reply:
<img width="812" alt="image" src="https://github.com/user-attachments/assets/0445f625-6403-41bf-9b75-dd85cbf7a2a7">
According to this image, this register is used for success/error of syscall. Note that the same register can be used for both input and outout.

Reply:
Oh I see it's the output register. We don't use that register for that purpose in cannon. So I would simply not introduce a new name for `RegSyscallError` to avoid any confusion. `RegA3` is fine to use elsewhere.

Reply:
How does cannon check syscall error then?

Reply:
I think it's used [here](https://github.com/ethereum-optimism/optimism/blob/c93c972a3d180ed5f8983f88e18f6d55584c8c32/cannon/mipsevm/exec/mips_syscalls.go#L285) and [here](https://github.com/ethereum-optimism/optimism/blob/c93c972a3d180ed5f8983f88e18f6d55584c8c32/cannon/mipsevm/multithreaded/mips.go#L63). Since `RegA3` services different purpose as input or output, I think it's reasonable to have two names referring to it?

Reply:
We discussed offline to have this renamed to `RegSyscallErrno`.